### PR TITLE
Merge Entities branch on Master

### PR DESCRIPTION
**Pull request by jbaptisteq**
All feats add.

To Do after merge : 
- Minor change graphic interface
- Update docs
- Final Fixtures

### CODE REVIEW

File: app/config/config_dev.yml [hunk 1]
```diff
@@ -38,4 +38,6 @@ monolog:
         #    level: info
 
 #swiftmailer:
```
Comment (Fschardt) [inline app/config/config_dev.yml]: Can be deleted if not used. No need to keep it commented

File: app/config/routing.yml [hunk 2]
```diff
@@ -1,3 +1,42 @@
-p6CoreBundle:
+index:
     resource: "@p6CoreBundle/Resources/config/routing.yml"
     prefix:   /
+
+homepage:
+    path:    /
```
Comment (Fschardt) [inline app/config/routing.yml]: Only one space after path:

File: app/config/routing.yml [hunk 3]
```diff
@@ -1,3 +1,42 @@
...
+    defaults: { _controller: p6CoreBundle:Index:index }
+
+register:
+    path:     /register
```
Comment (Fschardt) [inline app/config/routing.yml]: Only one space after path:

File: app/config/security.yml [hunk 4]
```diff
@@ -2,10 +2,28 @@
...
+
+
+    role_hierarchy:
+        ROLE_ADMIN:       ROLE_USER
+
     # https://symfony.com/doc/current/security.html#b-configuring-how-users-are-loaded
     providers:
-        in_memory:
-            memory: ~
+        #in_memory:
```
Comment (Fschardt) [inline app/config/security.yml]: Can be deleted. No need to keep it commented

File: src/p6/CoreBundle/Controller/IndexController.php [hunk 5]
```diff
@@ -6,21 +6,38 @@
...
+
+    /**
+     * @Route("/trick/detail/{id}", name="trick")
+     */
+    public function viewAction($id)
+    {
+      $entityManager = $this->getDoctrine()->getManager();
+      $trick = $entityManager->getRepository('p6CoreBundle:Trick')->find($id);
+
+      if(null === $trick){
```
Comment (Fschardt) [inline src/p6/CoreBundle/Controller/IndexController.php]: PSR2 : if (null === $trick) {

File: src/p6/CoreBundle/Resources/views/Index/index.html.twig [hunk 6]
```diff
@@ -3,31 +3,47 @@
...
         <div class="col-xs-2"><i class="far fa-trash-alt red"></i></div>
+        {% endif %}
     </div>
     {% endfor %}
 </div>
 
 <div class="container desktop">
     <div class="row">
-{% set nbtrick = 5 %}
+{% set nbtrick = 15 %}
```
Comment (Fschardt) [inline src/p6/CoreBundle/Resources/views/Index/index.html.twig]: So, what if we have more than 15 tricks to display ?

File: src/p6/UserBundle/Controller/RegistrationController.php [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+            $this->get('mailer')->send($message);
+
+            // 5) save the User!
+            $entityManager = $this->getDoctrine()->getManager();
+            $entityManager->persist($user);
+            $entityManager->flush();
+
+            return $this->redirectToRoute('homepage');
+            }
```
Comment (Fschardt) [inline src/p6/UserBundle/Controller/RegistrationController.php]: Rerun indentation on whole file. From line 67 there is one indentation level that should not exist

File: src/p6/UserBundle/Controller/SecurityController.php [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+class SecurityController extends Controller
+{
+  /**
+  * @Route("/login", name="login")
+  */
+  public function loginAction()
+  {
+    if ($this->get('security.authorization_checker')->isGranted('IS_AUTHENTICATED_REMEMBERED')){
```
Comment (Fschardt) [inline src/p6/UserBundle/Controller/SecurityController.php]: PSR2 : space before {

File: src/p6/UserBundle/Resources/config/services.yml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+services:
+#    p6_user.example:
```
Comment (Fschardt) [inline src/p6/UserBundle/Resources/config/services.yml]: To remove

**Review by Fschardt — CHANGES_REQUESTED** [review]
De manière générale, ne pas laisser de code commenté inutile, attention également aux lignes vides

File: app/config/routing.yml (hunk 2 above)
Comment (jbaptisteq) [reply app/config/routing.yml]: Code corrected

File: app/config/routing.yml (hunk 3 above)
Comment (jbaptisteq) [reply app/config/routing.yml]: Code corrected

File: app/config/security.yml (hunk 4 above)
Comment (jbaptisteq) [reply app/config/security.yml]: Code delete

File: src/p6/CoreBundle/Controller/IndexController.php (hunk 5 above)
Comment (jbaptisteq) [reply src/p6/CoreBundle/Controller/IndexController.php]: PSR2 apply

File: src/p6/CoreBundle/Resources/views/Index/index.html.twig (hunk 6 above)
Comment (jbaptisteq) [reply src/p6/CoreBundle/Resources/views/Index/index.html.twig]: line doesnt exist anymore. Replace by function.

File: src/p6/UserBundle/Controller/SecurityController.php (hunk 8 above)
Comment (jbaptisteq) [reply src/p6/UserBundle/Controller/SecurityController.php]: Corrected

File: src/p6/UserBundle/Resources/config/services.yml (hunk 9 above)
Comment (jbaptisteq) [reply src/p6/UserBundle/Resources/config/services.yml]: clear

File: src/p6/CoreBundle/Controller/TrickController.php [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+  /**
+  * @Route("/editTrick/{id}", name="editTrick")
+  */
+  public function editAction(Request $request, $id)
+  {
+    $entityManager = $this->getDoctrine()->getManager();
+    $trick = $entityManager->getRepository('p6CoreBundle:Trick')->find($id);
+    $form = $this->createForm(TrickType::class, $trick);
+
+    if(null === $trick){
```
Comment (Fschardt) [inline src/p6/CoreBundle/Controller/TrickController.php]: PSR2

Comment (Fschardt) [inline src/p6/CoreBundle/Controller/TrickController.php]: PSR2

File: src/p6/CoreBundle/Controller/TrickController.php [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+        $entityManager = $this->getDoctrine()->getManager();
+        $image = $entityManager->getRepository('p6CoreBundle:Image')->find($id);
+        $entityManager->remove($image);
+        $entityManager->flush();
+
+        $this->addFlash('success', 'Image Supprimée.');
+
+        return $this->redirectToRoute('trick', array('id' => $idtrick));
+      }
+      $this->addFlash('alert', 'Erreur lors de la suppréssion de l\'image.');
```
Comment (Fschardt) [inline src/p6/CoreBundle/Controller/TrickController.php]: Can be deleted with 143

File: src/p6/CoreBundle/Controller/TrickController.php [hunk 12]
```diff
@@ -0,0 +1,252 @@
...
+        $entityManager = $this->getDoctrine()->getManager();
+        $video = $entityManager->getRepository('p6CoreBundle:Video')->find($id);
+        $entityManager->remove($video);
+        $entityManager->flush();
+
+        $this->addFlash('success', 'Vidéo Supprimée.');
+
+        return $this->redirectToRoute('trick', array('id' => $idtrick));
+      }
+      $this->addFlash('alert', 'Erreur lors de la suppréssion de la vidéo');
```
Comment (Fschardt) [inline src/p6/CoreBundle/Controller/TrickController.php]: Can be deleted with 172

File: src/p6/CoreBundle/Controller/TrickController.php [hunk 13]
```diff
@@ -0,0 +1,252 @@
...
+
+        $newImage = $request->request->get('newImage');
+        $image->setUrl($newImage);
+        $entityManager->flush();
+
+        $this->addFlash('success', 'Image Editée.');
+
+        return $this->redirectToRoute('trick', array('id' => $idtrick));
+      }
+      $this->addFlash('alert', 'Erreur lors de l\'édition de l\'image.');
```
Comment (Fschardt) [inline src/p6/CoreBundle/Controller/TrickController.php]: Can be deleted with 203

File: src/p6/CoreBundle/Controller/TrickController.php [hunk 14]
```diff
@@ -0,0 +1,252 @@
...
+        $newLink = $request->request->get('newLink');
+        $video->setLink($newLink);
+
+        $entityManager->flush();
+
+        $this->addFlash('success', 'Vidéo Modifiée');
+
+        return $this->redirectToRoute('trick', array('id' => $idtrick));
+      }
+      $this->addFlash('alert', 'Erreur lors de la modification de la vidéo');
```
Comment (Fschardt) [inline src/p6/CoreBundle/Controller/TrickController.php]: Can be deleted with 235

**Review by Fschardt — CHANGES_REQUESTED** [review]

File: src/p6/CoreBundle/Controller/TrickController.php (hunk 10 above)
Comment (jbaptisteq) [reply src/p6/CoreBundle/Controller/TrickController.php]: Done

File: src/p6/CoreBundle/Controller/TrickController.php (hunk 11 above)
Comment (jbaptisteq) [reply src/p6/CoreBundle/Controller/TrickController.php]: Done

File: src/p6/CoreBundle/Controller/TrickController.php (hunk 12 above)
Comment (jbaptisteq) [reply src/p6/CoreBundle/Controller/TrickController.php]: Done

File: src/p6/CoreBundle/Controller/TrickController.php (hunk 13 above)
Comment (jbaptisteq) [reply src/p6/CoreBundle/Controller/TrickController.php]: Done

File: src/p6/CoreBundle/Controller/TrickController.php (hunk 14 above)
Comment (jbaptisteq) [reply src/p6/CoreBundle/Controller/TrickController.php]: Done

**Comment by jbaptisteq** [comment]
Correction  and Commit  done.

**Review by Fschardt — APPROVED** [review]

**Pull request merged**